File: 3.Upload_API/Upload_API_NET/Core/DTOs/ServiceResult.cs
using System;
using System.Collections.Generic;
using System.Net;
using System.Text;

namespace Core.DTOs
{
    public class ServiceResult
    {
        public bool Success { get; set; }
        public object Data { get; set; }
        public HttpStatusCode StatusCode { get; set; }
        public List<string> DevMsg { get; set; }
        public List<string> UserMsg { get; set; }
    }
}

File: 5.Identity-service_API/domain/schemas/service_result.py
from typing import Generic, TypeVar, Optional, Any

T = TypeVar('T')
class ServiceResult(Generic[T]):
    def __init__(
        self, 
        success: bool, 
        data: Optional[T] = None, 
        user_msg: Optional[str] = None, 
        dev_msg: Optional[str] = None,
        error_code: Optional[str] = None
    ):
        self.success = success       # Trạng thái True/False
        self.data = data             # Đối tượng trả về (User, Token, List...)
        self.user_msg = user_msg     # Thông báo thân thiện cho người dùng hiển thị lên UI
        self.dev_msg = dev_msg       # Thông báo chi tiết lỗi cho Developer/Log hệ thống
        self.error_code = error_code # Mã lỗi nội bộ hệ thống (Ví dụ: USER_LOCKED, PASS_INVALID)

